ts/Extensions/Boost/BoostSeries.ts

```diff
@@ -849,6 +859,111 @@ function getPoint(
     return point;
 }
 
+/**
+ * @private
+ */
+function scatterProcessData(
+    this: Series,
+    force?: boolean
+): (boolean|undefined) {
+    const series = this,
+        options = series.options,
+        xAxis = series.xAxis,
+        yAxis = series.yAxis;
+    (window as any).TheSeries = series;
+
+    // Process only on changes
+    if (
+        !series.isDirty &&
+        !xAxis.isDirty &&
+        !yAxis.isDirty &&
+        !force
+    ) {
+        return false;
+    }
+
+    // Required to get tick-based zoom ranges
+    series.yAxis.setTickInterval();
+
+    const boostThreshold = options.boostThreshold || 0,
+        cropThreshold = options.cropThreshold,
+        data = options.data || series.data,
+        xData = series.xData as Array<number>,
+        xExtremes = xAxis.getExtremes(),
+        xMax = xExtremes.max,
+        xMin = xExtremes.min,
+        yData = series.yData as Array<number>,
+        yExtremes = yAxis.getExtremes(),
+        yMax = yExtremes.max,
+        yMin = yExtremes.min;
+
+    // Skip processing in non-boost zoom
+    if (
+        !series.boosted &&
+        xAxis.old &&
+        yAxis.old &&
+        xMin >= (xAxis.old.min || 0) &&
+        xMax <= (xAxis.old.max || Number.MAX_VALUE) &&
+        yMin >= (yAxis.old.min || 0) &&
+        yMax <= (yAxis.old.max || Number.MAX_VALUE)
+    ) {
+        return true;
+    }
+
+    // Without thresholds just assign data
+    if (
+        (
+            boostThreshold &&
+            data.length < boostThreshold
+        ) ||
+        (
+            cropThreshold &&
+            !series.forceCrop &&
+            !series.getExtremesFromAll &&
+            !options.getExtremesFromAll &&
+            data.length < cropThreshold
+        )
+    ) {
+        series.processedXData = xData;
+        series.processedYData = yData;
+        return true;
+    }
+
+    // Filter unsorted scatter data for ranges
+    const processedData: Array<PointOptions> = [],
+        processedXData: Array<number> = [],
+        processedYData: Array<number> = [];
+
+    let cropped = false,
+        x: number,
+        y: number;
+
+    for (let i = 0, iEnd = xData.length; i < iEnd; ++i) {
+        x = xData[i];
+        y = yData[i];
+
+        if (
+            x >= xMin && x <= xMax &&
+            y >= yMin && y <= yMax
+        ) {
+            processedData.push({ x, y });
+            processedXData.push(x);
+            processedYData.push(y);
+        } else {
+            cropped = true;
+        }
+    }
+
+    // Set properties as base processData
+    series.cropped = cropped;
+    series.cropStart = 0;
+    series.processedData = processedData; // For un-boosted points rendering
+    series.processedXData = processedXData; // For boosted points rendering
+    series.processedYData = processedYData;
```

Comment:
Do we need to set both these varieties? It will double the memory use. I don't remember if there was a legit reason for storing both `processedData` and `processedXData`/`processedYData` at the same time. But if there is, we need a proper explanation in a comment.